### PR TITLE
Responde workspace configuration with workspace buffer

### LINKED ISSUE #2806: Workspace configuration response may not use file-local variables
## Description

Under some conditions, when `find-file` is not called interactively, `lsp-mode` will not use file-local variables while responding to workspace configuration, because the `current-buffer` haven't switched to the newly opened buffer yet.

---

## Possible fix

The following patch would fix this issue, but I'm not sure if it's OK so I didn't submit a PR

```diff
--- a/lsp-mode.el
+++ b/lsp-mode.el
@@ -5978,8 +5978,9 @@ WORKSPACE is the active workspace."
                                                     (error-message-string err))
                                         :json-false))))
                      ((equal method "workspace/configuration")
-                      (with-lsp-workspace workspace
-                        (lsp--build-workspace-configuration-response params)))
+                      (lsp-with-current-buffer (car buffers)
+                        (with-lsp-workspace workspace
+                          (lsp--build-workspace-configuration-response params))))
                      ((equal method "workspace/workspaceFolders")
                       (let ((folders (or (-> workspace
                                              (lsp--workspace-client)
```

---

## How to reproduce

(I encountered this issue while opening files using [libvterm message parsing](https://github.com/akermu/emacs-libvterm#message-passing). Here I provide a non-realistic but simpler way to reproduce.)

(Personally I use pyright to reproduce. But it should be the same for others.)

1. Install `lsp-pyright`. Add `lsp` to `prog-mode` hook. Enable `lsp-log-io`.
2. In `/tmp/py/.dir-locals.el` : `((python-mode . ((lsp-pyright-extra-paths . ["/some/extra/path"]))))`
3. In scratch buffer, run: `(run-with-timer 1 nil #'find-file "/tmp/py/hello.py")`

Expected behavior:

"/some/extra/path" should be sent in workspace configuration

Current behavioor:

"/some/extra/path" is not sent in workspace configuration

### PR DESCRIPTION
**Pull request by blahgeek**
Fixes #2806

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -5979,7 +5985,11 @@ WORKSPACE is the active workspace."
                                         :json-false))))
                      ((equal method "workspace/configuration")
                       (with-lsp-workspace workspace
-                        (lsp--build-workspace-configuration-response params)))
+                        (if-let ((buf (car buffers)))
+                            (with-current-buffer buf
```
Comment (yyoncho) [inline lsp-mode.el:5989]: can you use lsp-with-current-buffer?

**Review by yyoncho — COMMENTED** [review]
Looks good, one minor nit:

File: lsp-mode.el (hunk 1 above)
Comment (blahgeek) [reply lsp-mode.el]: fixed

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**